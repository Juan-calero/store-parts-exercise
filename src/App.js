import { BrowserRouter as Router, Route } from "react-router-dom";
import MainPage from "./components/MainPage";
import { bg, title } from "./app.module.scss";
import "./reset.scss";
import ProductPage from "./components/ProductPage";

function App() {
	return (
		<div className={bg}>
			<header>
				<h1 className={title}>Juan Calero Store Parts</h1>
			</header>
			<Router>
				<Route exact path="/">
					<MainPage />
				</Route>
				<Route path="/product/:productName" component={ProductPage} />
			</Router>
		</div>
	);
}

export default App;
