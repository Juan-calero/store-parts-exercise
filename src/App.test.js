import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import App from "./App";

Enzyme.configure({ adapter: new Adapter() });

test("Should match snapshot", () => {
	render(<App />);
	const tree = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});

test("H1 is displaying correctly", () => {
	const wrapper = shallow(<App />);
	const text = wrapper.find("h1.title");
	expect(text.text()).toBe("Juan Calero Store Parts");
});
