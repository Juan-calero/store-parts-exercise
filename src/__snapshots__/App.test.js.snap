// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  className="bg"
>
  <header>
    <h1
      className="title"
    >
      Juan Calero Store Parts
    </h1>
  </header>
  <main
    className="mainSection"
  >
    <form
      className="mainForm"
    >
      <input
        aria-label="Search"
        onChange={[Function]}
        placeholder="Search..."
        type="search"
        value=""
      />
      <select
        aria-label="Type"
        onChange={[Function]}
      >
        <option
          value="Type"
        >
          Type
        </option>
      </select>
      <input
        aria-label="Price Order"
        className="priceOrder"
        onClick={[Function]}
        type="button"
        value="Ascending Price"
      />
    </form>
    <section
      className="productSection"
    >
      <ul />
    </section>
  </main>
</div>
`;
